sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -280,8 +280,11 @@ object GpuDeviceManager extends Logging {
       }
 
       if (conf.isUvmEnabled) {
-        init = init | RmmAllocationMode.CUDA_MANAGED_MEMORY
         features += "UVM"
+        // Enable managed memory only if async allocator is not used.
+        if ((init | RmmAllocationMode.CUDA_ASYNC) == 0) {
+          init = init | RmmAllocationMode.CUDA_MANAGED_MEMORY
+        }
```

Comment:
else, I think we should throw. Not unlike: https://github.com/rapidsai/cudf/blob/branch-23.04/java/src/main/java/ai/rapids/cudf/Rmm.java#L226